classes/submission/SubmissionDAO.inc.php

```diff
@@ -94,30 +95,110 @@ function newDataObject() {
 	/**
 	 * Internal function to return a Submission object from a row.
 	 * @param $row array
+	 * @param $submissionRevision int optional
 	 * @return Submission
 	 */
-	function _fromRow($row) {
+	function _fromRow($row, $submissionRevision = 1) {
 		$submission = $this->newDataObject();
 
 		$submission->setId($row['submission_id']);
+		$submission->setSubmissionRevision($submissionRevision);
 		$submission->setContextId($row['context_id']);
 		$submission->setLocale($row['locale']);
 		$submission->setStageId($row['stage_id']);
 		$submission->setStatus($row['status']);
 		$submission->setSubmissionProgress($row['submission_progress']);
 		$submission->setDateSubmitted($this->datetimeFromDB($row['date_submitted']));
 		$submission->setDateStatusModified($this->datetimeFromDB($row['date_status_modified']));
-		$submission->setDatePublished($this->datetimeFromDB($row['date_published']));
```

Comment:
This is necessary, because the getDatePublished on a submission objects is often use. For all changes in pkp-lib we also have to keep OMP in mind. I would suggest keeping this and below in getById...